<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Construction of SCE object</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/spacelab.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/accessible-code-block-0.0.1/empty-anchor.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">ASTEC-sc</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="vignette.html">Tutorial</a>
</li>
<li>
  <a href="SCEconstruction.html">SCE object</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Construction of SCE object</h1>

</div>


<div id="build-your-sce-object" class="section level1">
<h1>Build your SCE object</h1>
<p>The principal steps to build the SingleCellExperiment (SCE) object required to use the ASTEC-sc shiny application is detailed in this section.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="kw">library</span>(SingleCellExperiment)</span></code></pre></div>
<p><strong>Step 1: Creation of SingleCellExperiment object</strong><br />
Let <code>X</code>, <code>Xnorm</code> and <code>Xlog</code> be a matrix of raw counts, normalized counts and lognormalized counts respectively. The SCE object is then initialized as follows :</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a>SCE &lt;-<span class="st"> </span><span class="kw">SingleCellExperiment</span>(<span class="dt">assays=</span><span class="kw">SimpleList</span>(<span class="dt">counts=</span>X,</span>
<span id="cb2-2"><a href="#cb2-2"></a>                                             <span class="dt">normcounts=</span>Xnorm,</span>
<span id="cb2-3"><a href="#cb2-3"></a>                                             <span class="dt">logcounts=</span>Xlog)) </span></code></pre></div>
<p>If you want to add another transformed data matrix <code>Xtranf</code>:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a><span class="kw">assays</span>(SCE, <span class="st">&quot;mytransf&quot;</span>) &lt;-<span class="st"> </span>Xtranf  <span class="co">#Optional</span></span></code></pre></div>
<p><strong>Step 2: Dimensionality reduction methods</strong></p>
<p>Let <code>RedDim1</code>, <code>RedDim2</code>, … be the coordinate matrices associated to several dimensionality reduction methods. You may enter them into the SCE object in the following way:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a><span class="kw">reducedDims</span>(SCE)&lt;-</span>
<span id="cb4-2"><a href="#cb4-2"></a><span class="st">  </span><span class="kw">SimpleList</span>(<span class="dt">NameRedDims1 =</span> RedDims1,</span>
<span id="cb4-3"><a href="#cb4-3"></a>             <span class="dt">NameRedDims2 =</span> RedDims2,</span>
<span id="cb4-4"><a href="#cb4-4"></a>             ...)</span></code></pre></div>
<p>At least one dimensionality reduction method is required in the SCE object.</p>
<p><strong>Step 3: cell clusterings</strong></p>
<p>Let <code>Clust1</code>, <code>Clust2</code>, … be vectors which contain a cell clustering respectively. To integrate them in the SCE object, it is required to enter a list of these cell clusterings in <code>SCE@metadata[["clustering"]]</code>:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a>SCE<span class="op">@</span>metadata[[<span class="st">&quot;clustering&quot;</span>]]&lt;-<span class="kw">list</span>(<span class="dt">nameClust1=</span>Clust1,</span>
<span id="cb5-2"><a href="#cb5-2"></a>                                  <span class="dt">nameClust2=</span>Clust2,</span>
<span id="cb5-3"><a href="#cb5-3"></a>                                            ...)</span></code></pre></div>
<p>At the end of this step, your SCE object is ready to be used in the application.</p>
<p><strong>Step 4: Add cell / feature information (optional)</strong></p>
<p>You may add supplementary information about cells and/or features in the SCE object.</p>
<p>If you have qualitative cell information (available in vectors <code>vector1</code>, <code>vector2</code>, …. respectively), you may save it in <code>SCE@metadata</code>. Be careful, your vectors must be of length the total number of cells in the SCE object (<code>length(vector1)==ncols(SCE)</code>).</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a>SCE<span class="op">@</span>metadata[[<span class="st">&quot;cellType1&quot;</span>]] &lt;-<span class="st"> </span>vector1</span>
<span id="cb6-2"><a href="#cb6-2"></a>SCE<span class="op">@</span>metadata[[<span class="st">&quot;cellType2&quot;</span>]] &lt;-<span class="st"> </span>vector2 </span></code></pre></div>
<p>If you have qualitative feature information (available in vectors <code>vectorfeature1</code>, <code>vectorfeature2</code>, …. respectively), you may save it in <code>SCE@int_elementmetadata</code>. You may integrate biological function information of some features in <code>knownfunc</code>, which must be a binary dataframe with some features in rows and functions in columns.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a>SCE<span class="op">@</span>int_elementMetadata<span class="op">$</span>NameType1 &lt;-<span class="st"> </span>vectorfeature1</span>
<span id="cb7-2"><a href="#cb7-2"></a>SCE<span class="op">@</span>int_elementMetadata<span class="op">$</span>NameType2 &lt;-<span class="st"> </span>vectorfeature2</span>
<span id="cb7-3"><a href="#cb7-3"></a>SCE<span class="op">@</span>int_elementMetadata<span class="op">$</span>KnownFunc &lt;-<span class="st"> </span>knownfunc</span></code></pre></div>
<p>To use your SCE object, you have to save it in a Rdata file:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a><span class="kw">save</span>(SCE,<span class="dt">file=</span><span class="st">&quot;mySCE.RData&quot;</span>)</span></code></pre></div>
</div>
<div id="our-sce-object-example" class="section level1">
<h1>Our SCE object example</h1>
<p>The construction of our SCE object example is here detailed. It is based on single cell RNA-seq data available in <code>scRNAseq</code> package and studied in <span class="citation">Zeisel et al. (2015)</span>.</p>
<p>Firstly, the object is initialized with the raw counts.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a><span class="kw">library</span>(scRNAseq)</span>
<span id="cb9-2"><a href="#cb9-2"></a>ZeiselBrain &lt;-<span class="st"> </span><span class="kw">ZeiselBrainData</span>()</span>
<span id="cb9-3"><a href="#cb9-3"></a><span class="kw">library</span>(SingleCellExperiment)</span>
<span id="cb9-4"><a href="#cb9-4"></a>SCE &lt;-<span class="st"> </span><span class="kw">SingleCellExperiment</span>(<span class="dt">assays=</span><span class="kw">SimpleList</span>(<span class="dt">counts=</span><span class="kw">counts</span>(ZeiselBrain)))</span>
<span id="cb9-5"><a href="#cb9-5"></a>genes=<span class="kw">rownames</span>(SCE)</span></code></pre></div>
<p>We remove genes that have an expression counts less than 25 for each cell.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a>delete&lt;-<span class="kw">which</span>(<span class="kw">apply</span>(<span class="kw">counts</span>(SCE),<span class="dv">1</span>,max)<span class="op">&lt;</span><span class="dv">25</span>)</span>
<span id="cb10-2"><a href="#cb10-2"></a>genes &lt;-<span class="st"> </span>genes[<span class="op">-</span>delete]</span>
<span id="cb10-3"><a href="#cb10-3"></a>SCE &lt;-<span class="st"> </span>SCE[<span class="op">-</span>delete,]</span></code></pre></div>
<p>We use the <code>scater</code> package to compute (log-transformed) normalized expression values:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a><span class="kw">library</span>(scater)</span>
<span id="cb11-2"><a href="#cb11-2"></a>SCE&lt;-<span class="kw">logNormCounts</span>(SCE,<span class="dt">log=</span>F) <span class="co">#normcounts</span></span>
<span id="cb11-3"><a href="#cb11-3"></a>SCE&lt;-<span class="kw">logNormCounts</span>(SCE,<span class="dt">log=</span>T) <span class="co">#logcounts</span></span></code></pre></div>
<p>The package <code>scater</code> allows us to easily build some dimensionality reduction methods for SingleCellExperiment object. See <a href="http://bioconductor.org/packages/release/bioc/vignettes/scater/inst/doc/overview.html#5_dimensionality_reduction">scater vignette</a> for more details.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a><span class="co"># PCA</span></span>
<span id="cb12-2"><a href="#cb12-2"></a>pcalogcounts &lt;-<span class="st"> </span><span class="kw">runPCA</span>(SCE,<span class="dt">ncomponents=</span><span class="dv">10</span>,<span class="dt">exprs_values=</span><span class="st">&quot;logcounts&quot;</span>) </span>
<span id="cb12-3"><a href="#cb12-3"></a>pcanormcounts &lt;-<span class="st"> </span><span class="kw">runPCA</span>(SCE,<span class="dt">ncomponents=</span><span class="dv">10</span>,<span class="dt">exprs_values=</span><span class="st">&quot;normcounts&quot;</span>)</span>
<span id="cb12-4"><a href="#cb12-4"></a><span class="co"># tSNE</span></span>
<span id="cb12-5"><a href="#cb12-5"></a>tsnelogcounts&lt;-<span class="kw">runTSNE</span>(SCE,<span class="dt">exprs_values =</span> <span class="st">&quot;logcounts&quot;</span>)</span>
<span id="cb12-6"><a href="#cb12-6"></a>tsnenormcounts&lt;-<span class="kw">runTSNE</span>(SCE,<span class="dt">exprs_values =</span> <span class="st">&quot;normcounts&quot;</span>)</span>
<span id="cb12-7"><a href="#cb12-7"></a><span class="co"># UMAP</span></span>
<span id="cb12-8"><a href="#cb12-8"></a>umaplogcounts&lt;-<span class="kw">runUMAP</span>(SCE,<span class="dt">exprs_values =</span> <span class="st">&quot;logcounts&quot;</span>)</span>
<span id="cb12-9"><a href="#cb12-9"></a>umapnormcounts&lt;-<span class="kw">runUMAP</span>(SCE,<span class="dt">exprs_values =</span><span class="st">&quot;normcounts&quot;</span>)</span></code></pre></div>
<p>You can enter them into the SCE in the following way.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a><span class="kw">reducedDims</span>(SCE)&lt;-</span>
<span id="cb13-2"><a href="#cb13-2"></a><span class="st">  </span><span class="kw">SimpleList</span>(<span class="dt">PCAlogcounts=</span><span class="kw">reducedDim</span>(pcalogcounts,<span class="st">&quot;PCA&quot;</span>),</span>
<span id="cb13-3"><a href="#cb13-3"></a>             <span class="dt">tSNElogcounts=</span><span class="kw">reducedDim</span>(tsnelogcounts,<span class="st">&quot;TSNE&quot;</span>),</span>
<span id="cb13-4"><a href="#cb13-4"></a>             <span class="dt">UMAPlogcounts=</span><span class="kw">reducedDim</span>(umaplogcounts,<span class="st">&quot;UMAP&quot;</span>),</span>
<span id="cb13-5"><a href="#cb13-5"></a>             <span class="dt">PCAnormcounts=</span><span class="kw">reducedDim</span>(pcanormcounts,<span class="st">&quot;PCA&quot;</span>),</span>
<span id="cb13-6"><a href="#cb13-6"></a>             <span class="dt">tSNEnormcounts=</span><span class="kw">reducedDim</span>(tsnenormcounts,<span class="st">&quot;TSNE&quot;</span>),</span>
<span id="cb13-7"><a href="#cb13-7"></a>             <span class="dt">UMAPnormcounts=</span><span class="kw">reducedDim</span>(umapnormcounts,<span class="st">&quot;UMAP&quot;</span>))</span></code></pre></div>
<p>We use the clustering available in <code>ZeiselBrain</code> and other cell clusterings are computed using <code>Seurat</code>, <code>pcaReduce</code> and <code>SC3</code>.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1"></a>Clust1 &lt;-<span class="st"> </span>ZeiselBrain<span class="op">@</span>colData<span class="op">$</span><span class="st">`</span><span class="dt">group #</span><span class="st">`</span></span>
<span id="cb14-2"><a href="#cb14-2"></a><span class="co">#pcaReduce</span></span>
<span id="cb14-3"><a href="#cb14-3"></a>pca.red &lt;-<span class="st"> </span><span class="kw">PCAreduce</span>(<span class="kw">t</span>(<span class="kw">logcounts</span>(SCE)), <span class="dt">nbt =</span> <span class="dv">3</span>, <span class="dt">q =</span> <span class="dv">20</span>, <span class="dt">method =</span> <span class="st">&#39;S&#39;</span>)[[<span class="dv">1</span>]]</span>
<span id="cb14-4"><a href="#cb14-4"></a>Clust2&lt;-pca.red[,<span class="dv">13</span>]</span>
<span id="cb14-5"><a href="#cb14-5"></a><span class="co"># SC3</span></span>
<span id="cb14-6"><a href="#cb14-6"></a>sceaux&lt;-SCE</span>
<span id="cb14-7"><a href="#cb14-7"></a><span class="kw">logcounts</span>(sceaux)&lt;-<span class="kw">as.matrix</span>(<span class="kw">logcounts</span>(sceaux))</span>
<span id="cb14-8"><a href="#cb14-8"></a><span class="kw">counts</span>(sceaux)&lt;-<span class="kw">as.matrix</span>(<span class="kw">counts</span>(sceaux))</span>
<span id="cb14-9"><a href="#cb14-9"></a><span class="kw">normcounts</span>(sceaux)&lt;-<span class="kw">as.matrix</span>(<span class="kw">normcounts</span>(sceaux))</span>
<span id="cb14-10"><a href="#cb14-10"></a><span class="kw">rowData</span>(sceaux)<span class="op">$</span>feature_symbol&lt;-<span class="kw">rownames</span>(sceaux)</span>
<span id="cb14-11"><a href="#cb14-11"></a>ressc3&lt;-<span class="kw">sc3</span>(sceaux,<span class="dt">ks=</span><span class="dv">9</span>,<span class="dt">gene_filter=</span>T,<span class="dt">n_cores=</span><span class="dv">2</span>)</span>
<span id="cb14-12"><a href="#cb14-12"></a>Clust3&lt;-<span class="kw">colData</span>(ressc3)[,<span class="st">&quot;sc3_9_clusters&quot;</span>]</span>
<span id="cb14-13"><a href="#cb14-13"></a><span class="co"># Seurat</span></span>
<span id="cb14-14"><a href="#cb14-14"></a>ZeiselBrainSeurat&lt;-<span class="kw">as.Seurat</span>(SCE,<span class="dt">counts =</span> <span class="st">&quot;counts&quot;</span>,<span class="dt">data =</span> <span class="st">&quot;logcounts&quot;</span>,<span class="dt">assay =</span> <span class="st">&quot;RNA&quot;</span>,<span class="dt">project =</span> <span class="st">&quot;SingleCellExperiment&quot;</span>)</span>
<span id="cb14-15"><a href="#cb14-15"></a>ZeiselBrainSeurat&lt;-<span class="kw">FindVariableFeatures</span>(ZeiselBrainSeurat, <span class="dt">selection.method =</span> <span class="st">&quot;vst&quot;</span>, <span class="dt">nfeatures =</span> <span class="dv">2000</span>)</span>
<span id="cb14-16"><a href="#cb14-16"></a>all.genes &lt;-<span class="st"> </span><span class="kw">rownames</span>(ZeiselBrainSeurat)</span>
<span id="cb14-17"><a href="#cb14-17"></a>ZeiselBrainSeurat &lt;-<span class="st"> </span><span class="kw">ScaleData</span>(ZeiselBrainSeurat, <span class="dt">features =</span> all.genes)</span>
<span id="cb14-18"><a href="#cb14-18"></a>ZeiselBrainSeurat &lt;-<span class="st"> </span><span class="kw">RunPCA</span>(ZeiselBrainSeurat, <span class="dt">features =</span> <span class="kw">VariableFeatures</span>(<span class="dt">object =</span> ZeiselBrainSeurat))</span>
<span id="cb14-19"><a href="#cb14-19"></a>ZeiselBrainSeurat &lt;-<span class="st"> </span><span class="kw">FindNeighbors</span>(ZeiselBrainSeurat, <span class="dt">dims =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">10</span>)</span>
<span id="cb14-20"><a href="#cb14-20"></a>ZeiselBrainSeurat &lt;-<span class="st"> </span><span class="kw">FindClusters</span>(ZeiselBrainSeurat, <span class="dt">resolution =</span> <span class="fl">0.5</span>)</span>
<span id="cb14-21"><a href="#cb14-21"></a>Clust4&lt;-<span class="kw">Idents</span>(ZeiselBrainSeurat)</span></code></pre></div>
<p>We integrate these cell clusterings in our SCE object as follows :</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1"></a>SCE<span class="op">@</span>metadata[[<span class="st">&quot;clustering&quot;</span>]]&lt;-<span class="kw">list</span>(<span class="dt">ClustZeisel=</span>Clust1,</span>
<span id="cb15-2"><a href="#cb15-2"></a>                                  <span class="dt">ClustPcared=</span>Clust2,</span>
<span id="cb15-3"><a href="#cb15-3"></a>                                  <span class="dt">ClustSC3=</span>Clust3,</span>
<span id="cb15-4"><a href="#cb15-4"></a>                                  <span class="dt">ClustSeurat=</span>Clust4)</span></code></pre></div>
<p>We integrate some cell information available in <code>ZeiselBrain</code> :</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1"></a>SCE<span class="op">@</span>metadata[[<span class="st">&quot;tissue&quot;</span>]]&lt;-ZeiselBrain<span class="op">@</span>colData<span class="op">$</span>tissue</span>
<span id="cb16-2"><a href="#cb16-2"></a>SCE<span class="op">@</span>metadata[[<span class="st">&quot;level1class&quot;</span>]]&lt;-ZeiselBrain<span class="op">@</span>colData<span class="op">$</span>level1class</span>
<span id="cb16-3"><a href="#cb16-3"></a>SCE<span class="op">@</span>metadata[[<span class="st">&quot;sex&quot;</span>]]&lt;-ZeiselBrain<span class="op">@</span>colData<span class="op">$</span>sex</span></code></pre></div>
<p>We integrate biological function information of some features in <a href="GenesFunctionZeisel.xlsx"><code>knownfunc</code></a>:</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1"></a><span class="kw">library</span>(readxl)</span>
<span id="cb17-2"><a href="#cb17-2"></a>knownFunc &lt;-<span class="st"> </span><span class="kw">read_excel</span>(<span class="st">&quot;GenesFunctionZeisel.xlsx&quot;</span>)</span>
<span id="cb17-3"><a href="#cb17-3"></a>knownFunc &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">left_join</span>(<span class="kw">data.frame</span>(genes), knownFunc, <span class="dt">by=</span><span class="kw">c</span>(<span class="st">&quot;genes&quot;</span>=<span class="st">&quot;Names&quot;</span>))</span>
<span id="cb17-4"><a href="#cb17-4"></a>knownFunc[<span class="kw">is.na</span>(knownFunc)] &lt;-<span class="st"> </span><span class="dv">0</span></span>
<span id="cb17-5"><a href="#cb17-5"></a></span>
<span id="cb17-6"><a href="#cb17-6"></a>SCE<span class="op">@</span>int_elementMetadata<span class="op">$</span>Name=genes</span>
<span id="cb17-7"><a href="#cb17-7"></a>SCE<span class="op">@</span>int_elementMetadata<span class="op">$</span>KnownFunc &lt;-<span class="st"> </span>knownFunc</span></code></pre></div>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1"></a><span class="kw">save</span>(SCE,<span class="dt">file=</span><span class="st">&quot;SCE-ZeiselBrain-example.RData&quot;</span>)</span></code></pre></div>
</div>
<div id="reference" class="section level1 unnumbered">
<h1>Reference</h1>
<div id="refs" class="references">
<div id="ref-zeisel2015cell">
<p>Zeisel, Amit, Ana B Muñoz-Manchado, Simone Codeluppi, Peter Lönnerberg, Gioele La Manno, Anna Juréus, Sueli Marques, et al. 2015. “Cell Types in the Mouse Cortex and Hippocampus Revealed by Single-Cell Rna-Seq.” <em>Science</em> 347 (6226): 1138–42.</p>
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
