<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Tutorial of ASTEC-sc</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/spacelab.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/accessible-code-block-0.0.1/empty-anchor.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>






<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-sm-12 col-md-4 col-lg-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-sm-12 col-md-8 col-lg-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">ASTEC-sc</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Overview</a>
</li>
<li>
  <a href="vignette.html">Tutorial</a>
</li>
<li>
  <a href="SCEConstruction.html">SCE object</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Tutorial of ASTEC-sc</h1>

</div>


<p>This vignette provide a quick-start guide using data from <span class="citation">Zeisel et al. (2015)</span> to illustrate the functionalities of the shiny application <a href="https://shiny-app-nec-okd46-2-shiny-nec.apps.math.cnrs.fr">ASTEC-sc</a>.<br />
You may download the SCE object to test the application here : <a href="SCE-ZeiselBrain-example.RData">SCE-ZeiselBrain-example.RData</a></p>
<div id="upload-data" class="section level1">
<h1>Upload data</h1>
<p>Upload data tab is a required way to use the app. For now, to use this application, you have to upload a SingleCellExperiment (SCE) object. This object has to be saved in a Rdata file. You can upload data and, when upload step is complete, a summary of the elements available in the SCE object is given.</p>
<p><img align="center" src="Pictures/UploadInput.png"></p>
<!--![](Pictures/UploadInput.png)-->
<p>This object must have a specific structure. You may refer to this <a href="SCEConstruction.html">vignette</a> to see how build a SCE object.</p>
<p>Please follow these recommendations for building your SCE object:</p>
<ul>
<li><code>sce@assays</code> contains the different types of data (raw counts (counts), normalized counts (normcounts), log-normalized counts (logcounts), …). By default: ‘counts’, ‘logcounts’ and ‘normcounts’ are required. The rows correspond to features (e.g. genes), the columns to cells.</li>
<li><code>sce@metadata</code>(optional) contains supplementary information for cells (qualitative variables).</li>
<li><code>sce@metadata$clustering</code> contains a list of the different cell clusterings.</li>
<li><code>reducedDims(sce)</code> contains a list of coordinate matrices for each considered dimensionality reduction method.</li>
<li><code>sce@int_elementMetadata</code> contains the supplementary information for features In particular, <code>sce@int_elementMetadata$KnownFunc</code> (optional) may contain a binary data frame where each column corresponds to a biological function and all features in rows.</li>
</ul>
<p>You may verify the structure of the SCE object with the following information described in the tab. This information will be shown when the object is uploaded. The summary of the SCE is composed of:</p>
<ul>
<li>the number of cells and features</li>
<li>the different types of data (contained in <code>sce@assays</code>)</li>
<li>the list of cell clusterings with the number of clusters</li>
<li>the list of dimensionality reduction methods with their number of dimensions</li>
<li>the list of cell information with their levels (optional)</li>
<li>the list of biological functions and the number of features (optional)</li>
</ul>
<p><img align="center" src="Pictures/UploadOutput1.png"></p>
<!-- ![](Pictures/UploadOutput1.png)-->
<p><img src="Pictures/UploadOutput2.png" /></p>
<p><img src="Pictures/UploadOutput3.png" /></p>
<p>If any information missed in the description, follow the recommendations on the <a href="SCEConstruction.html">vignette</a> for building your SCE object.</p>
</div>
<div id="descriptive-statistics" class="section level1">
<h1>Descriptive statistics</h1>
<p>The purpose of Descriptive statistics tab is to show the expression of data.</p>
<p>In this tab, you may play with two variables :</p>
<ol style="list-style-type: decimal">
<li>The “Choose the type of data” input allows you to choose between the different types of data (assays name). By default, the assays “counts” is chosen.</li>
<li>The “cell information” input allows you to choose between information contains in <code>sce@metadata</code> without clusterings (namely the supplementary information for cells), and the empty choice ‘–’ (by default).</li>
</ol>
<p><img src="Pictures/DescriptiveInput.png" /></p>
<p>The first plot shows the percent of expressed cells by mean of cell expression. The type of data input change the x-axis (the mean of cells expression).</p>
<p><img src="Pictures/DescriptiveOutput1.png" /></p>
<p>The two following plots show the proportion of expressed cells/expressed features respectively. You may plot result by cell information if you change the empty choice in the input.</p>
<p><img src="Pictures/DescriptiveOutput2.png" /></p>
<p>The last two plots show the density of mean expression by cells and features respectively. These plots are updated with the input values.</p>
<p><img src="Pictures/DescriptiveOutput3.png" /></p>
<p>A “Download Plot” button is available in this tab. If you click on, you can download all these plots in only a unique one png image.</p>
</div>
<div id="visualization-of-data" class="section level1">
<h1>Visualization of data</h1>
<p>This tab allows to visualize cells in a low-dimensional space based on the dimensionality reduction methods available in the SCE object.</p>
<p><img src="Pictures/Visualization1.png" /></p>
<p>For a great visualization, you can play with the three inputs on this tab.</p>
<ol style="list-style-type: decimal">
<li><p>The “Choose the coordinates of dimension reduction” input lets you to choose values contained in reducedDims(sce). By default, the first reduced dimension method of your object is selected.</p></li>
<li><p>The “Cell information for colour” input allows to colour cell points as follows:</p></li>
</ol>
<ul>
<li>‘–’ : all points will be black (by default)</li>
<li>cell information contained in <code>sce@metadata</code> with categorical colour</li>
<li>cell clusterings contained in <code>sce@metadata$clustering</code> with categorical colour</li>
<li>‘UMI’ is sequential colour for the sum of raw counts per cell: <span class="math inline">\(\forall c \in C, {UMI_c} =\sum_{g \in G} y_{gc}\)</span> where <span class="math inline">\(y_{gc}\)</span> is the expression counts of gene <span class="math inline">\(g\)</span> in cell <span class="math inline">\(c\)</span>.</li>
</ul>
<p>The “Choose the number of dimensions” depends on the dimensionality reduction input. You may choose from to until the number of dimensions of the selected method. If you choose n dimensions, a combination of the n first dimensions will be printed. We advise you to choose only two dimensions because of calculating time increasing with number of dimensions.</p>
<p><img src="Pictures/Visualization2.png" /></p>
<p>Only one plot is shown on “Visualization of data” tab. This is a graph of the cells in a low-dimensional space. With the package <code>plotly</code> to print the plot, you can zoom in to watch a part of data, select some categorical colour. Clicking on camera download a png image of your DT modification. You may also download the entire plot in a png image with the “Download Plot” button.</p>
</div>
<div id="comparison-of-several-cell-clusterings" class="section level1 tabset">
<h1>Comparison of several cell clusterings</h1>
<p>The “Comparison of several cell clusterings” tab allows to compare cell clusterings with two sub-tabs.</p>
<p>Inputs are common to both sub-tabs. The first, called “The main cell clusterings” is a unique choice selection input. You may choose a clustering from <code>sce@metadata$clustering</code>.<br />
The second input, called “The other cell clusterings” depends on the first input. You find, in the second input choices, the other cell clusterings from <code>sce@metadata$clustering</code>. You may choose a subset from the other clusterings.</p>
<p><img src="Pictures/ComparisonInput.png" /></p>
<div id="ari-sub-tab" class="section level2">
<h2>ARI sub-tab</h2>
<p>The ARI sub-tab plots a colour table of adjusted rand index (ARI) between each selected cell clustering. ARI is the corrected-for-chance version of a measure of the similarity between two cell clusterings. The closer ARI is to 1, the more similar the two clusterings are.</p>
<p><img src="Pictures/ComparisonARI.png" /></p>
<p>You may download this plot in a png image with the “Download Plot” button.</p>
</div>
<div id="circusplot-sub-tab" class="section level2">
<h2>CircusPlot sub-tab</h2>
<p><img src="Pictures/ComparisonCircusPlot.png" /></p>
<p>In the circus plot, choosing a main clustering makes sense. Indeed, this plot shows how many cells per cluster (from the main clustering) there are from other clusters (from other clusterings selected in the specific input). You may download this plot in a png image with the “Download Plot” button.</p>
</div>
</div>
<div id="chosen-cell-clustering" class="section level1">
<h1>Chosen cell clustering</h1>
<p>The “Chosen cell clustering” tab, composed of two sub-tab, focuses on a studied cell clustering.</p>
<p>Two inputs are common of the both sub-tabs. The “Choose the studied cell clustering” input allows to choose one cell clustering among those available in <code>sce@metadata$clustering</code>. All plots in this tab depend on this input. The “Choose an information for cells” input is the second common input. You may choose between cell information, contained in <code>sce@metadata$clustering</code> and the empty choice ‘–’.</p>
<p>The input “Choose a type of data” allows to choose one type of data name among those available in <code>assays(sce)</code>. This input is only used in ‘description’ sub-tab. The “Choose the studied cluster” input allows to choose one cluster among those available in the studied cell clustering. This input is only used in ‘Specific cluster’ sub-tab. The “Choose the dimensionality reduction method” input allows to choose between values contained in <code>reducedDims(sce)</code>. This input is only used in ‘Specific cluster’ sub-tab.</p>
<p><img src="Pictures/ClusteringInput.png" /></p>
<div id="description-sub-tab" class="section level2">
<h2>Description sub-tab</h2>
<p><img src="Pictures/ClusteringDescription1.png" /></p>
<p><img src="Pictures/ClusteringDescription2.png" /></p>
<p><img src="Pictures/ClusteringDescription3.png" /></p>
<p>The aim of “Description” sub-tab is to give information about one clustering and its clusters.</p>
<p>The first plot is a bar-plot about the number of cells per cluster. The second plot is a bar-plot about the proportion of cell information per cluster. To access to this plot, select other values of the cell information input than the empty choice ‘–’. The third plot in a violin plot about mean cell expressions per cluster. This plot depends on the type of data input.</p>
<p>All plots have a “Download plot” button to save them in a png image</p>
</div>
<div id="specific-cluster-sub-tab" class="section level2">
<h2>Specific cluster sub-tab</h2>
<p>The aim of “Specific cluster” sub-tab is to describe a cluster of a selected clustering.</p>
<p><img src="Pictures/ClusteringSpecific.png" /></p>
<p>The first plot is visible when ‘–’ is not selected (in the “Choose an information for cells” input). This is a reminder of proportion of cell information for the selected cluster. The second plot is a visualization of the selected cell cluster in a chosen low-dimensional space. All plots have a “Download plot” button to save plots in a png image</p>
</div>
</div>
<div id="detection-of-marker-genes" class="section level1">
<h1>Detection of marker genes</h1>
<p>“Detection of marker genes” tab allows you to determine marker genes (namely genes with a difference of expression) between two cell groups (a group may be a cluster or a merge of several clusters). Some indicators could help to select these genes. To find marker genes, you have to select some clusters to compare expressing values.</p>
<p>Firstly, you have to select a clustering in the corresponding input. By default, the first clustering in <code>sce@metadata$clustering</code> is selected.</p>
<p>Secondly, you may define the two groups of cells for calculating. The first input ‘Choose one (some) studied cluster(s)’ is the first cell group. You could select one cluster or merge some clusters (but not all) for the first group. The second group, with the “Choose the comparing group of clusters (one, some or all others)” input, can be composed with the remaining cluster. You could select one cluster or merge some clusters. If you want to select all others cluster, you may select “all others” choice.</p>
<p><img src="Pictures/FindMarkerInput1vsA.png" /></p>
<p>Then, you are ready to detect marker genes. Please, click on “Go - Compute indicators”.</p>
<p>The second part of input allows to filter genes thanks to the dynamic choice of thresholds on the following indicators:</p>
<ul>
<li><strong>pct.1</strong> (resp. <strong>pct.2</strong>) is the percentage of cells belonging to the first group (resp. the second group) where the gene is expressed (count &gt; 0). These indicators belong to [0,1].</li>
<li><strong>Delta</strong> is the difference between pct.1 and pct.2. This indicator belongs to [0,1].</li>
<li><strong>avg_logFC</strong> is the log of the ratio of the average expression between the two groups. A positive value indicats that the gene is more highly expressed in the first group than the second.</li>
<li><strong>AUC</strong> represents the area under the ROC curve and is between 0.5 and 1.</li>
<li><strong>p_val_adj</strong> is the adjusted p-value of Wilcoxon test, based on Bonferroni correction using all genes in the dataset. It belongs to [0,1]</li>
</ul>
<p><img src="Pictures/FindMarkerIndicatorInput.png" /></p>
<p>On “Find Marker Genes” sub-tab, you obtain a table of result with indicator values for each filtered marker gene. The interactive table only shows the first ten genes, by default. You may browse the other page or display more lines. The “Download table” button saves a csv data frame with all your filtered genes.</p>
<p><img src="Pictures/FindMarkerOutput1vsA.png" /></p>
<p>On “Indicator” sub-tab, you may find definitions of the indicators and density of this indicator. That may help you to choose thresholds of these indicators. By default, a ridges plot is used for graphs but you may also choose box-plot or violin plot.</p>
<p><img src="Pictures/FindMarkerIndicatorsOutput.png" /></p>
</div>
<div id="study-of-selected-genes" class="section level1">
<h1>Study of selected genes</h1>
<p>Before analyzing genes, you need to choose genes. Firstly, you have to choose the source of the list of genes:</p>
<ul>
<li>“from your object” is the choice by default which allows all genes contained in your SCE object.</li>
<li>If you compute marker genes in “Detection of marker genes” tab, you can also choose “From detection of marker genes tab”. With this choice, you may select all or some genes from this list.</li>
<li>If <code>sce@int_elementMetadata$KnownFunc</code> is not empty and is a binary table with different genes by row and different function by columns, you may select genes from a specific gene function.</li>
</ul>
<p>In the second input, you can choose if you want to analyse all genes from the chosen list or just a part of this list. If you want to select some genes, a multiple choice input is created and you may select almost 20 genes. Note: if you choose the list ‘From your object’ you cannot select all genes from this list.</p>
<p><img src="Pictures/StudyInputObject.png" /></p>
<p><img src="Pictures/StudyOutputObject.png" /></p>
<p><img src="Pictures/StudyOutputMarielleObject.png" /></p>
<p>The analysis of selected genes is made in two parts:</p>
<ul>
<li><p>the first part is a summary of plots in the second part. This plot shows the average log expression and the percent expressed for a couple of gene and cluster.</p></li>
<li><p>the second part details the expression of gene with two graphs per gene:</p>
<ul>
<li>The first is the density of logcount expressions of the selected genes by clustering to see in which clusters the gene is expressed.</li>
<li>The second graph is a visualization of cells in a two-dimensional space chosen with the input “Choose the dimensionality reduction method”, colouring by the logcount values of the selected gene.</li>
</ul></li>
</ul>
<p><img src="Pictures/StudyInputFunction.png" /></p>
<p><img src="Pictures/StudyOutputFunction.png" /></p>
<p><img src="Pictures/StudyOutputFunctionMarielle.png" /></p>
</div>
<div id="reference" class="section level1 unnumbered">
<h1>Reference</h1>
<div id="refs" class="references">
<div id="ref-zeisel2015cell">
<p>Zeisel, Amit, Ana B Muñoz-Manchado, Simone Codeluppi, Peter Lönnerberg, Gioele La Manno, Anna Juréus, Sueli Marques, et al. 2015. “Cell Types in the Mouse Cortex and Hippocampus Revealed by Single-Cell Rna-Seq.” <em>Science</em> 347 (6226): 1138–42.</p>
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
